Src/AutoFixture/ElementsBuilder.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Ploeh.AutoFixture.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Draws a random element from the given collection
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class ElementsBuilder<T> : ISpecimenBuilder
+    {
+        private readonly T[] elements;
+        private readonly RandomNumericSequenceGenerator sequence;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ElementsBuilder{T}"/> class.
+        /// </summary>
+        /// <param name="elements">The collection from which elements should be drawn from.
+        /// It must contain at least one element.</param>
+        public ElementsBuilder(IEnumerable<T> elements)
+        {
+            if (elements == null)
+                throw new ArgumentNullException("elements");
+
+            this.elements = elements.ToArray();
+
+            if (this.elements.Length < 1)
+                throw new ArgumentException("elements must contain at least one element.", "elements");
+
+            //The RandomNumericSequenceGenerator is only created for collections of minimum 2 elements
+            if (this.elements.Length > 1)
+                this.sequence = new RandomNumericSequenceGenerator(0, this.elements.Length - 1);
+        }
+
+        /// <summary>
+        /// Returns one of the element present in the collection given when the object was constructed.
+        /// </summary>
+        /// <param name="request">The request that describes what to create.</param>
+        /// <param name="context">Not used.</param>
+        /// <returns>
+        /// One of the element present in the collection given to the constructor if <paramref name="request"/>
+        /// is a request for <typeparamref name="T"/>; otherwise, a <see cref="NoSpecimen"/> instance.
+        /// </returns>
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (!typeof(T).Equals(request))
+#pragma warning disable 618
```

Comment:
I added the warning suppression in a subsequent commit (ed5c062), I initially used ``new NoSpecimen()``` which is apparently what it should be. I think the warning is fine as it is.

What got me confuses is the message associated with commit 5b459db6:

> For consistency's
sake, we ask contributors to adhere to the established standard of
returning 'new NoSpecimen(request)' even when everything works fine
without it. The design would be slightly simpler if Request was simply
omitted.

I'm not sure about the meaning of the second sentence: does it mean that I should use the old way or the new way :question: